### PR TITLE
Day off balances related changes

### PR DESCRIPTION
**Pull request by volodymyr-sokol**
## Changes
- Added Admin, Approver, Member roles
- Added balance calculation with the hardcoded logic
- Removed calendar entity and associated public holidays directly to it
- Added day off approving, possibility to adjust member start date etc

### CODE REVIEW

**Review by marcusn90 — APPROVED** [review]

File: application/workspace/models.py [hunk 1]
```diff
@@ -53,18 +66,53 @@ class WorkspaceInvitation(Base, Persistent):
...
+        result = 0
+        holiday_dates = set(map(lambda h: h.date, holidays))
 
-class Holiday(Base, Persistent):
-    __tablename__ = 'holiday'
+        date = start_date
+
+        while date <= end_date:
+            is_holiday = False
+            for holiday_date in holiday_dates:
```
Comment (nestorsokil) [inline application/workspace/models.py]: I think dates are hashable, so you can just do `is_holiday = date in holiday_dates` without the loop at all

File: application/workspace/models.py [hunk 2]
```diff
@@ -53,18 +66,53 @@ class WorkspaceInvitation(Base, Persistent):
...
+    date = db.Column(db.Date, nullable=False)
 
+    @staticmethod
+    def get_work_days_count(ws_id, start_date, end_date):
+        holidays = WorkspaceHoliday.query().\
+            filter(WorkspaceHoliday.date.between(start_date, end_date)).\
+            filter(WorkspaceHoliday.ws_id == ws_id).\
+            all()
+        result = 0
+        holiday_dates = set(map(lambda h: h.date, holidays))
```
Comment (nestorsokil) [inline application/workspace/models.py]: you can use `frozenset` to make it explicit that the whole object is immutable

File: application/workspace/models.py (hunk 1 above)
Comment (volodymyr-sokol) [reply application/workspace/models.py]: It worked, thanks!

File: application/workspace/models.py (hunk 2 above)
Comment (volodymyr-sokol) [reply application/workspace/models.py]: Needed `set` to deduplicate, but it's no longer needed.

**Pull request merged**